genie-web/src/main/java/com/netflix/genie/web/services/impl/JobDirectoryServerServiceImpl.java

```diff
@@ -185,11 +217,25 @@ public void serveResource(
         }
 
         if (jobStatus.isActive() && isV4) {
-            // This is where we should forward down the gRPC channel for live logs. Not yet supported.
-            response.sendError(
-                HttpStatus.NOT_FOUND.value(),
-                "Logs for V4 jobs are not yet available while the job is still running"
-            );
+
+            final Optional<JobDirectoryManifest> manifest = this.agentFileManifestService.getManifest(jobId);
+            if (!manifest.isPresent()) {
+                log.warn("Manifest not found for active job: {}", jobId);
+                response.sendError(HttpStatus.SERVICE_UNAVAILABLE.value(), "Could not load manifest for job: " + jobId);
```

Comment:
Wouldn't a 404 make more sense? or are you using 503 so that clients might retry? Looks like there's a `Retry-After` header we might want to add as well?

Reply:
Yes i was saving 404 for "this definitely does not exist", whereas here we don't know.